x-pack/test/functional/apps/ml/data_frame_analytics/custom_urls.ts

```diff
@@ -0,0 +1,203 @@
+        await ml.navigation.navigateToMl();
+        await ml.navigation.navigateToDataFrameAnalytics();
+        await ml.dataFrameAnalyticsTable.waitForAnalyticsToLoad();
+
+        await ml.testExecution.logTestStep('open the custom URLs tab in the edit job flyout');
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.closeEditJobFlyout();
+      });
+
+      it('adds a custom URL with query entities to Discover in the edit job flyout', async () => {
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.addDiscoverCustomUrl(testDiscoverCustomUrl);
+        await ml.dataFrameAnalyticsEdit.updateAnalyticsJob();
+      });
+
+      it('adds a custom URL to Dashboard in the edit job flyout', async () => {
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.addDashboardCustomUrl(testDashboardCustomUrl, {
+          index: 1,
+          url: `dashboards#/view/${testDashboardId}?_g=(filters:!(),time:(from:'$earliest$',mode:absolute,to:'$latest$'))&_a=(filters:!(),query:(language:kuery,query:'airline:\"$airline$\"'))`,
+        });
+        await ml.dataFrameAnalyticsEdit.updateAnalyticsJob();
+      });
+
+      it('adds a custom URL to an external page in the edit job flyout', async () => {
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.addOtherTypeCustomUrl(testOtherCustomUrl);
+        await ml.dataFrameAnalyticsEdit.updateAnalyticsJob();
+      });
+
+      it('tests other type custom URL', async () => {
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.testOtherTypeCustomUrlAction(2, testOtherCustomUrl.url);
+      });
+
+      it('edits other type custom URL', async () => {
+        const edit = {
+          label: `${testOtherCustomUrl.url} edited`,
+          url: `${testOtherCustomUrl.url}guide/index.html`,
+        };
+        await ml.dataFrameAnalyticsTable.openEditFlyout(dfaJobId);
+        await ml.dataFrameAnalyticsEdit.openEditCustomUrlsForJobTab(dfaJobId);
+        await ml.testExecution.logTestStep('edit the custom URL in the edit job flyout');
+        await ml.dataFrameAnalyticsEdit.editCustomUrl(2, edit);
+
+        await ml.testExecution.logTestStep('tests custom URL edit has been applied');
+        await ml.dataFrameAnalyticsEdit.testOtherTypeCustomUrlAction(2, edit.url);
```

Comment:
Should we test if the label has been updated correctly as well in `testOtherTypeCustomUrlAction`?